math/pythagoras.cpp

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have 3 point x, y, z.
+ * Also we created method int distance which takes points and returns distance.
+ */
+template <typename T> class point {
+ private:
+    T x;
+    T y;
+    T z;
+
+ public:
+    /**
+    * We have two constructor. First is for 3D point. it takes T type 3 value
+    * and save it to our calss. Second one takes only 2 T type value and third
+    * one is defoult 0 and save it into the class too.
+    */
+    point(T a, T b, T c) : x(a), y(b), z(c) {}  // Consructor for 3D points
```

Comment:
in C++11, initializers can be combined with one initializer as such:
```cpp
point(T a, T b, T c=0) : x(a), y(b), z(c) {}  
```
wherein, the third parameter has a default value of zero and becomes an optional input argument.

Reply:
What do you suggest

Reply:
keeping just one initializer.